### DISCUSSION TITLE
Release and update .exe (nsis) & .msi packages simultaneously

### DISCUSSION THREAD

**Discussion by MrJamesT** [tauri-apps/tauri#8963] [Q&A] [answered]
Hi, I can't find the answer to this anywhere in the docs, what's the best way to do this? I published my package originally with NSIS and I feel like I'm getting less antivirus detections on MSI so I would like to at least start making .msi packages as well. However I also use the updater and I can only specify one target for the **platforms - windows-x86_64** json object. Do I need to have basically two versions of the app so that each one can have a different url for the manifest?

**Comment by FabianLars** [comment, marked as answer]
> Do I need to have basically two versions of the app so that each one can have a different url for the manifest?

For now yes. While you can upgrade from msi to nsis apps, the other way around is not supported.

If you don't use the default updater dialog (which will be removed in v2 anyway) you could dynamically set the endpoint url in rust though detecting how the app was installed is not straightforward. There should be registry entries to identify it but don't know them for sure rn.

**Comment by LickABrick** [comment]
Hi, was wondering what the current state of this is? How can I update using the msi when the user installed via msi and update with the exe when the user installed via exe?

**Reply by FabianLars** [reply]
you currently have to compile the app twice, changing the updater config for each so you can use 2 different updater endpoints. Since we now also support multiple bundle types on Linux i opened a feature request to discuss the approach here https://github.com/tauri-apps/plugins-workspace/issues/2277

----

if you don't mind sharing, what's your use-case for distributing both, the msi and nsis exe?

**Reply by LickABrick** [reply]
I'm not a programmer by profession so there might be other possibilities, but I use the .exe for a per-user install (no admin rights required by end user) and msi for when I want a per-machine install.

**Reply by FabianLars** [reply]
Ah alright that makes sense I guess. The nsis exe supports a `both` mode that allows per user and machine wide installs but to start the installer it always requires admin permissions no matter which mode ends up being used.
